### PR TITLE
Breaking changes

### PR DESCRIPTION
**Pull request by himedlooff**
- Updated to use cf-core which has some basic form styling that needed to be cleared out from here.
- Some formatting fixes like removing unnecessary prefixes and updating syntax.
- Adds some new patterns from cfgov-refresh.

Also let's chat about the custom form elements. My opinion is that each of them should have their own repos. The custom checkboxes and radio in one and the custom select in another. They are two different jQuery plugins that we'll need to write tests and I'd rather do that in separate repos than one large one.

### CODE REVIEW

**Comment by ascott1** [comment]
+1 to separate repos for the checkboxes/radio and select boxes.

**Comment by himedlooff** [comment]
Thanks for the feedback @ascott1!

**Comment by Scotchester** [comment]
Definitely agree on the separate repos.

File: src/cf-forms.less [hunk 1]
```diff
@@ -5,464 +5,291 @@
...
       notes:
-        - Please review the notes on form input success icons in the form icons section below.
-      markup: |
-        <input class="success" placeholder="placeholder search text" type="search" value="Validated input">
-        <i class="cf-form_input-icon icon-ok-sign">
-          <span class="jekyll-bug"></span>
-        </i>
-    tags:
+        - "When nesting a form input the input will receive a right margin to
+          space it from the label text."
```
Comment (Scotchester) [inline src/cf-forms.less]: It's worth pointing out that text inputs can be legally nested inside of labels, as well, and we may not (almost certainly would not) want this same behavior. If we want to discourage that pattern, we should make a note of that, as well.

File: src/cf-forms.less [hunk 2]
```diff
@@ -5,464 +5,291 @@
...
-input[type="search"] {
-    &:extend(input[type="text"]);
-}
-
-input[type="search"]:focus,
-input[type="search"].focus {
-    // Class included for static state examples
-    &:extend(input[type="text"]:focus);
-}
+.form-label {
```
Comment (Scotchester) [inline src/cf-forms.less]: Why not just apply these base styles to all `label` elements?

File: src/cf-forms.less [hunk 3]
```diff
@@ -5,464 +5,291 @@
...
-}
+.@{cf-forms_input-icon-class} {
+    @font-size: 20px;
 
-input[type="number"]:focus,
-input[type="number"].focus {
-    // Class included for static state examples
-    &:extend(input[type="text"]:focus);
+    position: relative;
+    top: unit( 3px / @font-size, em);
```
Comment (Scotchester) [inline src/cf-forms.less]: I request bumping this to `4px`. I'd rather the circular validation icons be vertically centered than the email icon. I think the validation icons will be the predominant use case.

File: src/cf-forms.less (hunk 1 above)
Comment (himedlooff) [inline src/cf-forms.less]: These patterns are coming from cfgov-refresh and I'm trying to port some patterns over. We can leave some out if we think they're not ready. I think maybe we can hold this one back.

File: src/cf-forms.less (hunk 2 above)
Comment (himedlooff) [inline src/cf-forms.less]: As per above I think we can hold this one back.

File: src/cf-forms.less (hunk 3 above)
Comment (himedlooff) [inline src/cf-forms.less]: Oh right, especially now with the larger input height.

Comment (himedlooff) [inline src/cf-forms.less]: Done, thanks!

File: src/cf-forms.less (hunk 2 above)
Comment (himedlooff) [inline src/cf-forms.less]: Held back.

**Comment by Scotchester** [comment]
All good now; thanks!

**Pull request merged**